services/expiration/expiration-notification.server.ts

```diff
@@ -186,8 +189,9 @@ const getExpiringDnsRecords = () => {
   });
 };
 // function to add jobs
-async function addExpirationNotifications() {
-  let jobName = `${expirationNotificationQueueName}`;
+async function addExpirationNotifications(type: string) {
+  // we never want to add multiple jobs for the same user/email notification (unique jobId)
```

Comment:
NOTE: the job name and id are not the same thing.  You need to put a unique `jobId` on the notification jobs themselves.  I gave you some code already in Slack:

```ts
export const addNotification = (data: NotificationData) => {
  const jobName = `notification`;
  data.jobId = `${data.emailAddress}:${data.subject}`;
  return notificationsQueue.add(jobName, data);
};
```

The name of the job doesn't affect duplication of jobs in the queue; only the `jobId` can fix that.

Reply:
Sorry, I am probably having a bad day. Thanks for reminding me!